src/nwbinspector/tools/_read_nwbfile.py

```diff
@@ -66,8 +70,40 @@ def read_nwbfile(
     nwbfile_path: Union[str, Path],
     method: Optional[Literal["local", "fsspec", "ros3"]] = None,
     backend: Optional[Literal["hdf5", "zarr"]] = None,
-    return_io: bool = False,
-) -> Union[NWBFile, tuple[NWBFile, HDMFIO]]:
+) -> NWBFile:
+    """
+    Read an NWB file using the specified (or auto-detected) method and specified (or auto-detected) backend.
+
+    Parameters
+    ----------
+    nwbfile_path : str or pathlib.Path
+        Path to the file on your system.
+    method : "local", "fsspec", "ros3", or None (default)
+        Where to read the file from; a local disk drive or steaming from an https:// or s3:// path.
+        The default auto-detects based on the form of the path.
+        When streaming, the default method is "fsspec".
+        Note that "ros3" is specific to HDF5 backend files.
+    backend : "hdf5", "zarr", or None (default)
+        Type of backend used to write the file.
+        The default auto-detects the type of the file.
+
+    Returns
+    -------
+    nwbfile : pynwb.NWBFile
+        The in-memory NWBFile object.
+    """
+    nwbfile, _ = _read_nwbfile_helper(nwbfile_path=nwbfile_path, method=method, backend=backend)
+
+    # Note: do not be concerned about IO object closing due to garbage collection here
+    # (the IO object is attached as an attribute to the NWBFile object)
+    return nwbfile
+
+
+def read_nwbfile_and_io(
+    nwbfile_path: Union[str, Path],
+    method: Optional[Literal["local", "fsspec", "ros3"]] = None,
+    backend: Optional[Literal["hdf5", "zarr"]] = None,
+) -> tuple[NWBFile, HDMFIO]:
```

Comment:
This was a really bad design pattern on my part, I and should have known that  the time

Mypy particularly complained about the split typing, so I've properly separated it into two public methods that both call the same private helper